crates/renderer/src/renderer.rs

```diff
@@ -83,21 +83,23 @@ pub fn render_skia(
             }
         }
 
+        let snapped_area = area.round();
```

Comment:
Does it really make sense to round these areas? For example, browsers do not round up paddings

Reply:
It's definetly a sensible choice for `rect`, although it might not be necessary for other element types:

(without `snapped_area`)
![image](https://github.com/marc2332/freya/assets/42101043/99425efd-9748-4b45-877e-97e6cb16103b)
![image](https://github.com/marc2332/freya/assets/42101043/8203d10c-5690-4136-a8b9-43b92a142a6e)

(with `snapped_area`)
![image](https://github.com/marc2332/freya/assets/42101043/de567ab0-1d50-4434-b14b-48d033ccbf41)
![image](https://github.com/marc2332/freya/assets/42101043/89fa28a3-61d3-4e79-973a-2091515133bb)

This seems to also have a *very slightly* improved effect on image rendering as well, but only on the edges in some rare cases. For the other elements, it's probably not a bad idea to round either way for edge cases like text decorations and SVG rendering.

Reply:
I'll see what I can do because I am getting some weird paddings on some components like the Switch

Reply:
Yeah,  vertical centering in places like switches can cause issues. This is a problem in browsers too iirc, and requires either having blurry edges for the switch handle or even spacing between the switch height and switch handle height.

Reply:
> Yeah, vertical centering in places like switches can cause issues. This is a problem in browsers too iirc, and requires either having blurry edges for the switch handle or even spacing between the switch height and switch handle height.

2.6px
![image](https://github.com/marc2332/freya/assets/38158676/a26b4f16-34e8-4962-9ecc-2c60a87d266b)

3px
![image](https://github.com/marc2332/freya/assets/38158676/38611976-9f63-4a7e-9cb4-7b91492fe647)


comparison 2.6px vs  3px

![image](https://github.com/marc2332/freya/assets/38158676/17891ec1-db02-4ceb-80b4-1edb9e63325a)

Mmm, do browsers actually round the pixels? This doesn't feel like they do, or at least, the same way you do